File: Algorithms/Medium/RemovingStarsFromAString.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
    def removeStars(self, s: str) -> str:
        stack = []

        for ch in s:
            if ch == "*":
                stack.pop()
            else:
                stack.append(ch)

        return "".join(stack)

File: Algorithms/Medium/2390.Removing-Stars-From-A-String/RemovingStarsFromAString.go
package medium

// Time complexity: O(n)
// Space complexity: O(n)
func removeStars(s string) string {
	stack := make([]byte, 0, len(s))
	pop := func() {
		if len(stack) > 0 {
			stack = stack[:len(stack)-1]
		}
	}

	for i := 0; i < len(s); i++ {
		if s[i] == '*' {
			pop()
		} else {
			stack = append(stack, s[i])
		}
	}

	return string(stack)
}
